common/dynamicconfig/constants.go

```diff
@@ -522,10 +522,14 @@ const (
 	HistoryShutdownDrainDuration = "history.shutdownDrainDuration"
 	// XDCCacheMaxSizeBytes is max size of events cache in bytes
 	XDCCacheMaxSizeBytes = "history.xdcCacheMaxSizeBytes"
-	// EventsCacheMaxSizeBytes is max size of events cache in bytes
+	// EventsCacheMaxSizeBytes is max size of the host level events cache in bytes
 	EventsCacheMaxSizeBytes = "history.eventsCacheMaxSizeBytes"
+	// EventsShardLevelCacheMaxSizeBytes is max size of the shard level events cache in bytes
+	EventsShardLevelCacheMaxSizeBytes = "history.eventsShardLevelCacheMaxSizeBytes"
```

Comment:
I'd keep the config values backward compatible, i.e. define a new config for host level cache max size instead of changing the meaning of the existing one. Otherwise upon rollout this will be a problem.

Reply:
I followed how we made mutable state cache host level. 
In that case, we created a new config for shard level cache here: https://github.com/temporalio/temporal/blob/a8f48d53617cf3bf509af26283fa37e99c00098f/common/dynamicconfig/constants.go#L515


I agree with you that this will cause problems on rollout. But maybe it is better to keep this behaviour consistent between both the caches?

Reply:
> we created a new config for shard level cache
🤦 @yux0 Can you confirm? That will be a problem for rollout right? We have an override for `history.cacheMaxSize` I believe.

> it is better to keep this behaviour consistent
Safety upon rollout & correctness is top priority I think. They are more important than consistent behavior. and I think we need to fix the config for history cache...

Reply:
Got it! I thought it is too late to fix that for mutable state cache. That is why I suggested we keep this like this.

Reply:
Changed it to be backward compatible.